static/js/echo.js

```diff
@@ -368,7 +368,7 @@ function abort_message(message) {
 
 exports.initialize = function () {
     function on_failed_action(action, callback) {
-        $("#main_div").on("click", "." + action + "-failed-message", function (e) {
+        $("#main_div").on("click", `.${CSS.escape(action)}-failed-message`, function (e) {
```

Comment:
Thoughts on just doing this?:

~~~ diff
--- a/static/js/echo.js
+++ b/static/js/echo.js
@@ -530,8 +530,8 @@ $(function () {
         preprocessors: [preprocess_code_blocks],
     });

-    function on_failed_action(action, callback) {
-        $("#main_div").on("click", "." + action + "-failed-message", function (e) {
+    function on_failed_action(selector, callback) {
+        $("#main_div").on("click", selector, function (e) {
             e.stopPropagation();
             popovers.hide_all();
             var row = $(this).closest(".message_row");
@@ -547,9 +547,9 @@ $(function () {
         });
     }

-    on_failed_action('remove', abort_message);
-    on_failed_action('refresh', resend_message);
-    on_failed_action('edit', edit_failed_message);
+    on_failed_action('.remove-failed-message', abort_message);
+    on_failed_action('.refresh-failed_message', resend_message);
+    on_failed_action('.edit-failed-message', edit_failed_message);

Reply:
Sure.

Reply:
Oops, I was on an old branch trying to track down a commit.  But it's the same idea on master (without `.edit-failed-message`).